graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -264,7 +265,12 @@ class DgsSchemaProvider(
                 .forEach { method ->
                     val mergedAnnotations =
                         MergedAnnotations.from(method, MergedAnnotations.SearchStrategy.TYPE_HIERARCHY)
-                    mergedAnnotations.stream(DgsData::class.java).forEach { dgsDataAnnotation ->
+                    val filteredMergedAnnotations =
+                        mergedAnnotations
+                            .stream(DgsData::class.java)
+                            .filter { (it.source as Method).declaringClass == method.declaringClass }
```

Comment:
I'm not sure we want to just check if the duplicate data fetchers are in the same class. That would avoid cases where we actually have duplicates (non-mocks).

Reply:
Perhaps we should filter out mocks instead, if possible.

Reply:
> Perhaps we should filter out mocks instead, if possible.

a vice versa - we need mock themselves, because we mock datafetchers in order to get new (provided by mock) response

Reply:
Sorry, I was not clear. What I mean is we allow duplicates only if the other data fetchers are mocks.

Reply:
> I'm not sure we want to just check if the duplicate data fetchers are in the same class. That would avoid cases where we actually have duplicates (non-mocks).

I don't understand your point =( I added that check because `MergedAnnotations.from(method, MergedAnnotations.SearchStrategy.TYPE_HIERARCHY)` returns two annotations - annotation belonging to method of real class and annotation belonging to method of generated mock.
Although at step before `ReflectionUtils.getUniqueDeclaredMethods(javaClass, ReflectionUtils.USER_DECLARED_METHODS` returns **only** Method belonging to generated mock.

So doing filtering I filter out excess annotations of "real method"

Reply:
What I mean is the following desired behavior:
1. If we have multiple data fetchers, check if the duplicates include a mock
2. Throw an exception only when we have multiple real data fetchers
3. Disable the check and allow it if there is one real and the rest are mocks

Hope that clarifies things a bit better.

Reply:
One more question, so If I understood you correctly, you want to allow duplicates in case mocks.

Can this lead us to errors or any inconsistency in future, due to having duplicates when we run test and not having them in normal execution ?

Reply:
I think we should be ok. Based on the above conditional checks, only mock duplicates are allowed. So for tests that rely on real data fetchers, they should not be setting up mocks anyway.

Reply:
The issue with duplicates is that it is not guaranteed which data fetcher will be picked, btw. So tests need to be aware of that if mocking data fetchers. 

The other approach is to be strict like before and not even allow mocks to guarantee full consistency. That would be my strong preference generally.  But as you described in your use case, if your test set up requires mocks in addition to real data fetchers, this is the next best solution I can think of.

Reply:
Done @srinivasankavitha

Reply:
Thanks for the changes and iterating on this. I see the check for Mockito mock in the class name. I'm still not convinced this is needed though, and feels like we are unnecessarily introducing checks for the sake of tests. 

How is your test set up actually? Are you able to use the real data fetcher and just mock the calls within the data fetcher at a lower level?

I think we really do want to keep the duplicate data fetcher check across different classes, so ideally, it is the test that should be set up to accomadate this behavior.

Reply:
> feels like we are unnecessarily introducing checks for the sake of tests.

My previous solution was without those checks) And it should work correctly, because
> Although at step before ReflectionUtils.getUniqueDeclaredMethods(javaClass, ReflectionUtils.USER_DECLARED_METHODS returns only Method belonging to generated mock. So doing filtering I filter out excess annotations of "real method"


> How is your test set up actually?
It's described here https://github.com/Netflix/dgs-framework/issues/1214#issuecomment-1236009171

> Are you able to use the real data fetcher and just mock the calls within the data fetcher at a lower level? 
Also there https://github.com/Netflix/dgs-framework/issues/1214#issuecomment-1236009171

Reply:
Yeah, I did go through those comments earlier. It was not clear to me why you could not mock the calls inside the data fetcher (e.g. if you are making a call to a service etc.) and let all the data fetchers execute per normal.

In any case, I think after the discussion and the changes, your original changes might be better than checking for mocks.

Reply:
Thank you too)

So, speaking about the fix, what is my following step ? Should I remove the second commit ? Or will you just apply the first one ? Or something else ?